lib/pos_random_seed.go

```diff
@@ -93,20 +103,30 @@ func (bav *UtxoView) VerifyRandomSeedSignature(
 	if err != nil {
 		return nil, errors.Wrapf(err, "UtxoView.VerifyRandomSeedSignature: problem retrieving CurrentRandomSeedHash: ")
 	}
-	isVerified, err := signerPublicKey.Verify(randomSeedSignature, currentRandomSeedHash[:])
+	randomSeedHash, err := verifyRandomSeedSignatureGivenRandomSeedHash(signerPublicKey, randomSeedSignature, currentRandomSeedHash)
 	if err != nil {
 		return nil, errors.Wrapf(err, "UtxoView.VerifyRandomSeedSignature: problem verifying RandomSeedSignature: ")
 	}
+	return randomSeedHash, nil
+}
+
+func verifyRandomSeedSignatureGivenRandomSeedHash(
+	signerPublicKey *bls.PublicKey, randomSeedSignature *bls.Signature, randomSeedHash *RandomSeedHash,
+) (*RandomSeedHash, error) {
+	isVerified, err := signerPublicKey.Verify(randomSeedSignature, randomSeedHash[:])
+	if err != nil {
+		return nil, errors.Wrapf(err, "UtxoView.verifyRandomSeedSignatureGivenRandomSeedHash: problem verifying RandomSeedSignature: ")
+	}
```

Comment:
See suggestion above on splitting the existing `verifyRandomSeedSignatureGivenRandomSeedHash` into two
```suggestion
func hashRandomSeedSignature(randomSeedSignature *bls.Signature) (*RandomSeedHash, error) {
	// This function takes in a random seed signature and computes the random seed hash for it
```